packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2377,10 +2417,12 @@ class _RouteEntry extends RouteTransitionRecord {
     }
 
     if (previousState == _RouteLifecycle.replace || previousState == _RouteLifecycle.pushReplace) {
+      navigator._inheritedObserver?.didReplace(newRoute: route, oldRoute: previousPresent);
```

Comment:
Maybe define an observers getter on the NavigatorState that returns the widget.observers followed by the inherited one and use that everywhere instead of `navigator.widget.observers` to avoid this duplication?

Reply:
This should have minimum duplication once https://github.com/flutter/flutter/pull/57605 is merged, I separate them out because i don't want to create a list or using a yield everytime the getter is called